src/components/utils/include/utils/timer.h

```diff
@@ -141,42 +141,30 @@ class Timer {
     bool stop_flag_;
 
     /*
-     * State lock used to protect condition variable
+     * Termination lock used to protect condition variable
      */
-    sync_primitives::Lock state_lock_;
+    sync_primitives::Lock termination_lock_;
     sync_primitives::ConditionalVariable termination_condition_;
 
     DISALLOW_COPY_AND_ASSIGN(TimerDelegate);
   };
 
-  void StopUnsafe();
+  void StopLockFree();
```

Comment:
@anosach-luxoft so may be we also need re-name the opposite `Stop` into `StopWithLock`